### PR TITLE
feat: update provider status when provider emits events

### PR DESCRIPTION
**Pull request by federicobond**
I also made `openfeature.provider.registry` a private module and moved the registry singleton inside it.

### CODE REVIEW

File: openfeature/provider/_registry.py [hunk 1]
```diff
@@ -74,31 +74,78 @@ def _initialize_provider(self, provider: FeatureProvider) -> None:
...
             else:
-                self._set_provider_status(provider, ProviderStatus.ERROR)
+                self.dispatch_event(
+                    provider,
+                    ProviderEvent.PROVIDER_ERROR,
+                    ProviderEventDetails(
+                        message=f"Provider initialization failed: {err}",
+                        error_code=ErrorCode.GENERAL,
+                    ),
+                )
```
[the remark this answers is not in the document]
Comment (federicobond) [reply openfeature/provider/_registry.py]: I've simplified the expression to always pass the error code from the OpenFeatureError.

[the remark this answers is not in the document]
Comment (gruebel) [reply openfeature/provider/_registry.py]: even better 😄

**Review by gruebel — APPROVED** [review]

**Comment by federicobond** [comment]
@gruebel found a couple of bugs with the event logic around provider initialization/shutdown, all new changes are in the latest commit.

File: openfeature/api.py [hunk 2]
```diff
@@ -10,15 +10,13 @@
 from openfeature.exception import GeneralError
 from openfeature.hook import Hook
 from openfeature.provider import FeatureProvider
+from openfeature.provider._registry import default_registry
```
Comment (beeme1mr) [inline openfeature/api.py:13]: Could we call this provider_registery? I don't find default_registery that intuitive unless we add a nondefault_registery.

**Review by beeme1mr — APPROVED** [review]
Left a suggestion but overall looks good.

File: openfeature/api.py (hunk 2 above)
Comment (federicobond) [reply openfeature/api.py:13]: Great point! I was also feeling a bit awkward about the name.

**Comment by gruebel** [comment]
@federicobond new changes look good 🙂

**Pull request merged**